### ISSUE TITLE
 camera models 

### ISSUE THREAD

**Issue by willer94** [MrNeRF/gaussian-splatting-cuda#45]
Hi, are your going to support more camera models in  future? such as opencv model.

**Comment by MrNeRF** [comment]
Hey, no current plans at the moment as I am too busy. However, maybe the opencv model. That one should be easy

**Comment by MrNeRF** [comment]
We should support it. For reference, https://github.com/nerfstudio-project/nerfstudio/blob/main/nerfstudio/process_data/colmap_utils.py#L187-L387

**Comment by altineller** [comment]
Hello,

I stumbled upon the same issue, as it rejected my colmap output for being wrong camera model. I found out that the code in colmap_reader.cpp around line 317 is responsible for camera models.

How hard can it be to implement this? Appearently gsplat supports it.

**Comment by MrNeRF** [comment]
Hey @altineller,

I would consider this rather an easy problem to be solved. You can look up [here](https://github.com/nerfstudio-project/gsplat/blob/main/examples/datasets/colmap.py) how that's done!

I am building currently around other issues. That's why it hasn't been done yet. However, any external contribution makes me very happy :)

**Comment by altineller** [comment]
yeah, ok, i think i can figure out this out. i am an experienced software developer, but a new comer in the field. i just need someone to ask a lot of questions, and i think i can contribute. will work on it on the weekend. best regards from istanbul. :)

**Comment by altineller** [comment]
Hello,

I have added:

	case CAMERA_MODEL::SIMPLE_RADIAL: {

           // f, cx, cy, k
           // r2 = u**2 + v**2;
           // radial = k * r2
           // du = u * radial
           // dv = u * radial

           float fx = out[i]._params[0].item<float>();
           float fy = out[i]._params[0].item<float>();
           float cx = out[i]._params[1].item<float>();
           float cy = out[i]._params[2].item<float>();
           float cz = out[i]._params[3].item<float>();

	   //out[i]._cx = cx;
	   //out[i]._cy = cy;
	   //out[i]._cz = cz;
	   //out[i]._k2 = 0.0;
	   //out[i]._p1 = 0.0;
	   //out[i]._p2 = 0.0

           out[i]._fov_x = focal2fov(fx, out[i]._width);
           out[i]._fov_y = focal2fov(fy, out[i]._height);
           break;

	}

since simple radial has cx cy cz and k2, aren't we supposed to process the image with these values? I have checked that the struct CameraData does not contain cx,cy, etc. 

Also there are two mentions of PINHOLE camera type in rasterizer_autograd. Do these need to change to SIMPLE_RADIAL when using simple_radial mode?

Right now, I am rendering my dataset after I made these changes, and it seems it is rendering, but maybe we are not taking full advantage of simple_radial parameters.

I have migrated to latest version with the vcpkg, and at the beginning it caoused some problems, but now it is working much nicely. How is the work on 3DGUT going?

Best Regards

**Comment by MrNeRF** [comment]
Hey,
Everything you need is outlined here https://github.com/nerfstudio-project/gsplat/blob/main/examples/datasets/Colman.py as linked above. `Simple_Radial has only the k1 coefficient and you need to apply the undistortion. By following the referenced Colmap importer you should be able to implement the cam type in this code base. Otherwise you need to wait until this is added which will happen when I am back in 2-3 weeks.

Hope this helps.
